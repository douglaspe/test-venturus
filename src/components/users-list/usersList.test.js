import React from 'react'
import { shallow } from 'enzyme';
import axios from 'axios';

import UsersList from './usersList';

describe('<UsersList />', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<UsersList />));

    it('should render <tr> Component', () => {
        //expect(wrapper.find('tr').length).toEqual(1);
    });

    const API = 'https://jsonplaceholder.typicode.com/users';

    it('should async API call', async () => {
        const req = await axios.get(API);
        const underTest = req.data[0];
        expect(underTest.name).toBe('Leanne Graham');
    });

}); 