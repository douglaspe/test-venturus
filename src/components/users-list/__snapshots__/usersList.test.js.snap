// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsersList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          Bret
        </td>,
        <td>
          Leanne Graham
        </td>,
        <td>
          Sincere@april.biz
        </td>,
        <td>
          Gwenborough
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bret",
        },
        "ref": null,
        "rendered": "Bret",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Leanne Graham",
        },
        "ref": null,
        "rendered": "Leanne Graham",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sincere@april.biz",
        },
        "ref": null,
        "rendered": "Sincere@april.biz",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gwenborough",
        },
        "ref": null,
        "rendered": "Gwenborough",
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            Bret
          </td>,
          <td>
            Leanne Graham
          </td>,
          <td>
            Sincere@april.biz
          </td>,
          <td>
            Gwenborough
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bret",
          },
          "ref": null,
          "rendered": "Bret",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Leanne Graham",
          },
          "ref": null,
          "rendered": "Leanne Graham",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sincere@april.biz",
          },
          "ref": null,
          "rendered": "Sincere@april.biz",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gwenborough",
          },
          "ref": null,
          "rendered": "Gwenborough",
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
